File: app/src/main/java/cn/intret/app/picgo/model/DetectFileExistenceResult.java
package cn.intret.app.picgo.model;


import java.io.File;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class DetectFileExistenceResult {
    /**
     * Key: folder
     * Value: existed files
     */
    Map<File, List<File>> mExistedFiles = new LinkedHashMap<>();

    public Map<File, List<File>> getExistedFiles() {
        return mExistedFiles;
    }

    public DetectFileExistenceResult setExistedFiles(Map<File, List<File>> existedFiles) {
        mExistedFiles = existedFiles;
        return this;
    }

    @Override
    public String toString() {
        return "DetectFileExistenceResult{" +
                "mExistedFiles=" + mExistedFiles +
                '}';
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/image/data/DetectFileExistenceResult.kt
package cn.intret.app.picgo.model.image.data


import java.io.File
import java.util.*

class DetectFileExistenceResult {
    /**
     * Key: folder
     * Value: existed files
     */
    var existedFiles: LinkedHashMap<File, List<File>>? = LinkedHashMap()


    fun setExistedFiles(existedFiles: LinkedHashMap<File, List<File>>?): DetectFileExistenceResult {
        this.existedFiles = existedFiles
        return this
    }

    override fun toString(): String {
        return "DetectFileExistenceResult{" +
                "existedFiles=" + existedFiles +
                '}'.toString()
    }

    init {
        existedFiles = LinkedHashMap()
    }
}
